### PR TITLE
Create cmake toolchain file and use it to set some sane values

### PR DESCRIPTION
**Pull request by wiiaboo**
Any way to make this a little better, @1480c1 ? I think you're way more comfortable with cmake than I am.

### CODE REVIEW

**Comment by 1480c1** [comment]
I can try, although I may need a bit of time to see exactly what is needed here

**Comment by wiiaboo** [comment]
It did fix the libtiff "issue" with Cmake trying to find msys2 libm, at least. Didn't test with any other cmake lib to see if anything broke.

File: build/media-suite_helper.sh [hunk 1]
```diff
@@ -1743,6 +1743,27 @@ EOF
             cat > "$LOCALDESTDIR"/bin/ab-pkg-config-static.bat
 }
 
+create_cmake_toolchain() {
+    local _win_path_LOCALDESTDIR="$(cygpath -m $LOCALDESTDIR)"
+    local _win_path_MINGW_PREFIX="$(cygpath -m $MINGW_PREFIX)"
+    local toolchain_file=(
+        "SET(CMAKE_C_COMPILER gcc)"
```
Comment (1480c1) [inline build/media-suite_helper.sh]: Is it necessary to hardcode gcc? Maybe better to let cmake detect c compiler/use CC variable

Comment (wiiaboo) [reply build/media-suite_helper.sh]: Sure, those three lines were sort of copied from https://github.com/shinchiro/mpv-winbuild-cmake/blob/master/toolchain.cmake.in

Comment (1480c1) [reply build/media-suite_helper.sh]: I see, I see

File: build/media-suite_helper.sh [hunk 2]
```diff
@@ -1743,6 +1743,25 @@ EOF
...
 }
 
+create_cmake_toolchain() {
+    local _win_path_LOCALDESTDIR="$(cygpath -m $LOCALDESTDIR)"
+    local _win_path_MINGW_PREFIX="$(cygpath -m $MINGW_PREFIX)"
+    local toolchain_file=(
+        "SET(CMAKE_RC_COMPILER_INIT windres)"
+        "SET(PKG_CONFIG_EXECUTABLE $_win_path_LOCALDESTDIR/bin/ab-pkg-config-static.bat)"
+        ""
+        "LIST(APPEND CMAKE_PROGRAM_PATH $_win_path_LOCALDESTDIR/bin ...)"
```
Comment (1480c1) [inline build/media-suite_helper.sh]: What's with the triple dot?

Comment (wiiaboo) [reply build/media-suite_helper.sh]: Probably shitty copy/paste from docs and thinking they do something like in JS.

**Comment by 1480c1** [comment]
Other than the potential problem of quotes around the paths, I don't see anything wrong/needing to change

**Comment by wiiaboo** [comment]
Real big brain stuff, didn't realize I'd pushed the commit to master already.

**Comment by 1480c1** [comment]
Lul

**Pull request merged**